### ISSUE TITLE
Value returned by amplify-appsync-simulator for AWSJSON fields doesn't match AppSync

### ISSUE THREAD

**Issue by ElvenMonky** [aws-amplify/amplify-cli#5207]
Currently 'amplify-appsync-simulator' uses 'graphql-type-json' to represent AWSJSON type. 'graphql-type-json' sets `identity` as a function to parse and serialize value.

When you query field of AWSJSON type from AppSync, response contains string value with serialized JSON.
'amplify-appsync-simulator' returns parsed value (object in my use case) in same situation.

**Amplify CLI Version**
I'm using "amplify-appsync-simulator@1.23.2" through "serverless-appsync-simulator" and don't have "amplify-cli" installed.

**To Reproduce**
Test AppSync locally with schema, that contains fields of AWSJSON type.

**Expected behavior**
Strings with serialized JSON are returned to client for AWSJSON fields.

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version. v12.18.0

**Comment by akshbhu** [comment]
Hi @ElvenMonky 

Thanks for reporting this.

Are you mutating with stringified JSON  in both Appsync console and appsync mock?

**Comment by ElvenMonky** [comment]
@akshbhu I think I need to describe my use case a bit more.

I have following relevant piece in "schema.graphql":
```
type Query {
  entity: EntityType;
}

type EntityType {
  id: ID!
  data: AWSJSON;
  timeStamp: AWSDateTime
}
```

I have HTTP datasource defined and resolver for "Query.entity" field that uses this HTTP datasource to call some rest API endpoint.

I use Serverless Framework with "serverless-appsync-plugin" and "serverless-appsync-simulator" plugins both to deploy to AWS AppSync and to run AppSync emulator locally. "serverless-appsync-simulator" configures and uses "amplify-appsync-simulator" under the hood.
So I don't mock any data in both cases, either when deployed to AppSync or when starting "AmplifyAppSyncSimulator" locally.

The difference is that deployed AppSync does necessary transformations for fields of AWS* types before returning to clients.
When running locally, no transformations are performed leading to not valid GraphQL responses.

I'm pretty sure it's relevant not only to my setup.

I did some more investigation and I see that if only scalar type definitions from [appsync-scalars](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-appsync-simulator/src/schema/appsync-scalars/index.ts) were somehow registered in [graphQL](https://github.com/graphql/graphql-js), then it has all code necessary to transform values before returning to client ([completeLeafValue](https://github.com/graphql/graphql-js/blob/d6e760cf7060217bd25fa934bd104ff2400aad96/src/execution/execute.js#L935))

Unfortunately, I don't know yet how to register type definitions in gpaphql-js and if it's even possible.

**Comment by ElvenMonky** [comment]
Basically I believe that example from [AWS Documentation](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html) (see "Example Schema Usage" section) is not going to work with "amplify-appsync-simulator".

**Comment by bboure** [comment]
Hi there,

I came across the same issue. 

I thought this was a bug in AppSync. I genuinely thought that the AWSJSON would return an actual **parsed** json.
Exemple shows that it should be a string. 
So I guess the bug is indeed here. 

[This line](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-appsync-simulator/src/schema/appsync-scalars/index.ts#L143) should change to a custom scalar that receives a string or an object, and checks that it is a valid json, then returns it back stringified. 

Does that sound right?

**Comment by edwardfoyle** [comment]
The fix for this has been released in CLI v 4.32.0
